src/creature.cpp

```diff
@@ -1233,6 +1233,17 @@ const effect &Creature::get_effect( const efftype_id &eff_id, body_part bp ) con
     }
     return effect::null_effect;
 }
+std::vector<effect *> Creature::get_all_effects_of_type( const efftype_id &eff_id )
+{
+    std::vector< effect *> ret;
+    auto got_outer = effects->find( eff_id );
+    if( got_outer != effects->end() ) {
+        for( auto &pr : got_outer->second ) {
+            ret.push_back( &pr.second );
+        }
+    }
```

Comment:
```suggestion
    if( got_outer == effects->end() ) {
        return {};
    }
    auto effect_map = got_outer->second;
    std::vector<const effect *> ret;
    for( const auto &[ _, effect ] : got_outer->second ) {
        ret.push_back( &effect );
    }
```
1. early return is preferred. the brace means it's returning [default initialized (empty) vector.](https://stackoverflow.com/a/39487231/13503626)
2. [pair unpacking](https://en.cppreference.com/w/cpp/language/structured_binding) is preferred for better readability.

Reply:
Personally find this somewhat less comprehensible than the original, but I don't know code.

Reply:
maybe it's personal preference, but i suck at comprehending what's `pr` and what's `pr->second`, to the point a spend a few minutes searching through the code what do they mean. i'd rather like them to have explicit name.

Reply:
I'll get used to it I'm sure.